btk/metrics.py

```diff
@@ -1,484 +1,351 @@
+            row["distance_detection"] = matches[i]["dist"][j]
+
+            # obtain distance to closest galaxy in the blend
+            if len(blend) > 1:
+                dists = []
+                for g in blend:
+                    dx = gal["x_peak"] - g["x_peak"]
+                    dy = gal["y_Peak"] - g["y_peak"]
+                    dists.append(np.hypot(dx, dy))
+                row["distance_closest_galaxy"] = np.partition(dists, 1)[1]
+            else:
+                row["distance_closest_galaxy"] = -1  # placeholder
+
+            row["blend_id"] = i
+            row["blendedness"] = get_blendedness(isolated_images[i][j], isolated_images[i])
+            if "reconstruction" in use_metrics:
+                row["mse"] = results["reconstruction"]["mse"][i][j]
+                row["psnr"] = results["reconstruction"]["psnr"][i][j]
+                row["ssim"] = results["reconstruction"]["ssim"][i][j]
+            if "segmentation" in use_metrics:
+                row["iou"] = results["segmentation"]["iou"][i][j]
+            results["galaxy_summary"].add_row(row[0])
 
-    Args:
-        metrics_params: Instance from class
-        `btk.metrics.Metrics_params` describing functions to return
-        results of detection/deblending/measurement algorithm.
-        test_size(int): Number of times Metrics_params is run and results
-            summarized.
+    return results
 
-    Returns:
-        dict summarizing detection/deblending/measurement results.
 
-    """
-    results = {
-        "detection": [astropy.table.Table(), astropy.table.Table(), []],
-        "segmentation": [],
-        "flux": [],
-        "shapes": [],
-    }
-    for i in range(test_size):
-        print(f"Running test {i}")
-        # Evaluate detection algorithm
-        try:
-            batch_detection_result = metrics_params.get_detections()
-        except GeneratorExit as e:
-            print(e)
-            print("GeneratorExit encountered. Returning results")
-            return results
-        if (
-            len(batch_detection_result[0]) != len(batch_detection_result[1])
-            or len(batch_detection_result[0]) != metrics_params.batch_size
-        ):
-            raise ValueError(
-                "Metrics_params.get_detections output must be "
-                "two lists of astropy table of length batch size."
-                f" Found {len(batch_detection_result[0])}, "
-                f"{len(batch_detection_result[1])}, "
-                f"{metrics_params.batch_size}"
+class MetricsGenerator:
+    """Generator that calculates metrics on batches returned by the MeasureGenerator."""
+
+    def __init__(
+        self, measure_generator, use_metrics=("detection"), meas_band_num=0, target_meas={}
+    ):
+        self.measure_generator = measure_generator
+        self.use_metrics = use_metrics
+        self.meas_band_num = meas_band_num
+        self.target_meas = target_meas
```

Comment:
What is `target_meas` ? I actually couldn't figure it out from looking at its usage.

Reply:
Basically, we want it to contain functions that measure a physical parameter on the images, which is then both ran on isolated_images and on deblended_images. Later, we can compare the two results and it gives an idea on how well the deblender performed relatively to the physical parameter you are interested in (currently only ellipticity but the structure is flexible)

Reply:
That makes sense, I added info in the docstring. Thanks!